Crypto/Random/Types.hs

```diff
@@ -25,6 +30,26 @@ class DRG gen where
     -- | Generate N bytes of randomness from a DRG
     randomBytesGenerate :: ByteArray byteArray => Int -> gen -> (byteArray, gen)
 
+-- | A Psuedo Random Generator (PRG) class
+--
+-- Like 'DRG' but also supports initialisation from some fixed seed.
+class DRG gen => PRG gen where
+    -- | Initialize the DRG from some fixed seed.
+    prgNewSeed :: ByteArrayAccess seed => seed -> gen
+    -- | Length of seed in bytes
+    prgSeedLength :: Proxy gen -> Int
```

Comment:
prgSeedLength could use a type variable `proxy` instead of concrete `Proxy`, this is more general, and ensures even better that the argument is never pattern-matched.

Reply:
Done in 1686aee